src/System.Private.CoreLib/shared/System/Enum.cs

```diff
@@ -288,94 +159,33 @@ private static string InternalFlagsFormat(RuntimeType eT, TypeValuesAndNames ent
-                    break;
-                case TypeCode.Int32:
-                    result = (ulong)(int)value;
-                    break;
-                case TypeCode.UInt64:
-                    result = (ulong)value;
-                    break;
-                case TypeCode.Int64:
-                    result = (ulong)(long)value;
-                    break;
-                // All unsigned types will be directly cast
-                default:
-                    throw new InvalidOperationException(SR.InvalidOperation_UnknownEnumType);
+                string[] names = Enum.InternalGetNames(eT);
+                return names[index];
             }
 
-            return result;
+            return null; // return null so the caller knows to .ToString() the input
         }
 
-        [MethodImplAttribute(MethodImplOptions.InternalCall)]
-        private static extern int InternalCompareTo(object o1, object o2);
-
-        [MethodImplAttribute(MethodImplOptions.InternalCall)]
-        internal static extern RuntimeType InternalGetUnderlyingType(RuntimeType enumType);
-
-        [DllImport(JitHelpers.QCall, CharSet = CharSet.Unicode)]
-        private static extern void GetEnumValuesAndNames(RuntimeTypeHandle enumType, ObjectHandleOnStack values, ObjectHandleOnStack names, bool getNames);
-
-        [MethodImplAttribute(MethodImplOptions.InternalCall)]
-        private static extern object InternalBoxEnum(RuntimeType enumType, long value);
-        #endregion
-
-        #region Public Static Methods
-        public static object Parse(Type enumType, string value) =>
-            Parse(enumType, value, ignoreCase: false);
-
-        public static object Parse(Type enumType, string value, bool ignoreCase)
+        internal static ulong[] InternalGetValues(RuntimeType enumType)
         {
-            bool success = TryParse(enumType, value, ignoreCase, throwOnFailure: true, out object result);
-            Debug.Assert(success);
-            return result;
+            // Get all of the values
+            return GetCachedValuesAndNames(enumType, false).Values;
         }
 
-        public static TEnum Parse<TEnum>(string value) where TEnum : struct =>
-            Parse<TEnum>(value, ignoreCase: false);
-
-        public static TEnum Parse<TEnum>(string value, bool ignoreCase) where TEnum : struct
+        internal static string[] InternalGetNames(RuntimeType enumType)
         {
-            bool success = TryParse<TEnum>(value, ignoreCase, throwOnFailure: true, out TEnum result);
-            Debug.Assert(success);
-            return result;
+            // Get all of the names
+            return GetCachedValuesAndNames(enumType, true).Names;
         }
 
-        public static bool TryParse(Type enumType, string value, out object result) =>
-            TryParse(enumType, value, ignoreCase: false, out result);
-
-        public static bool TryParse(Type enumType, string value, bool ignoreCase, out object result) =>
```

Comment:
Is there a intention behind some of this reordering? I think the code was much easier to understand when all Parse methods were together.